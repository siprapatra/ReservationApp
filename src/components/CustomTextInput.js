import React from "react";
import { TextInput, StyleSheet } from "react-native";
import PropTypes from "prop-types";

const CustomTextInput = ({ hint, onTextChange, type, value }) => (
  <TextInput
    placeholder={hint}
    style={styles.itemTextInput}
    value={value}
    onChangeText={text => onTextChange(text, type)}
  />
);

CustomTextInput.propTypes = {
  hint: PropTypes.string,
  onTextChange: PropTypes.func,
  type: PropTypes.string,
  value: PropTypes.string
};

CustomTextInput.defaultProps = {
  hint: "",
  onTextChange: () => {},
  type: "",
  value: ""
};

const styles = StyleSheet.create({
  itemTextInput: {
    borderColor: "black",
    borderWidth: 1,
    borderRadius: 5,
    padding: 10,
    margin: 10,
    height: 50,
    fontSize: 17,
    color: "black"
  }
});

export default CustomTextInput;
