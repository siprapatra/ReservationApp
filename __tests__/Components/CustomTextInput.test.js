import React from "react";
import renderer from "react-test-renderer";
import CustomTextInput from "../../src/components/CustomTextInput";

describe("Test for Custom textinput", () => {
  test("should render the textinput", () => {
    const snap = renderer
      .create(
        <CustomTextInput
          hint="Enter your arrival date"
          type="some type"
          onTextChange={(text)=>console.log(text)}
        />
      )
      .toJSON();
    expect(snap).toMatchSnapshot();
  });
  test("should render the textinput without props", () => {
    const snap = renderer
      .create(
        <CustomTextInput
        />
      )
      .toJSON();
    expect(snap).toMatchSnapshot();
  });
});
