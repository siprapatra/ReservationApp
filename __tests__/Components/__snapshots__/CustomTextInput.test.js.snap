// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for Custom textinput should render the textinput 1`] = `
<TextInput
  allowFontScaling={true}
  onChangeText={[Function]}
  placeholder="Enter your arrival date"
  rejectResponderTermination={true}
  style={
    Object {
      "borderColor": "black",
      "borderRadius": 5,
      "borderWidth": 1,
      "color": "black",
      "fontSize": 17,
      "height": 50,
      "margin": 10,
      "padding": 10,
    }
  }
  underlineColorAndroid="transparent"
  value=""
/>
`;

exports[`Test for Custom textinput should render the textinput without props 1`] = `
<TextInput
  allowFontScaling={true}
  onChangeText={[Function]}
  placeholder=""
  rejectResponderTermination={true}
  style={
    Object {
      "borderColor": "black",
      "borderRadius": 5,
      "borderWidth": 1,
      "color": "black",
      "fontSize": 17,
      "height": 50,
      "margin": 10,
      "padding": 10,
    }
  }
  underlineColorAndroid="transparent"
  value=""
/>
`;
